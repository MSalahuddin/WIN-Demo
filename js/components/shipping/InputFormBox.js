import React, { useState } from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { Platform } from 'react-native';
import Text, { SIZE, FONT_FAMILY } from '../common/Text';
import { scaleHeight, scaleWidth, scale } from '../../platformUtils';
import { color, fontFamily } from '../../styles';
import { error as errorIcon, tick } from '../../../assets/images';

const InputBox = styled.TextInput`
  border-color: ${({ borderColor }) => borderColor};
  border-width: ${({ borderWidth }) => borderWidth};
  border-radius: ${scaleHeight(5)};
  color: ${color.lightBlack};
  background-color:${color.white}
  font-size: ${scale(14)};
  height: ${Platform.OS === 'android' ? scaleHeight(50) : scaleHeight(35)};
  padding-left: ${scaleWidth(8)};
  padding-top: ${Platform.OS === 'android' ? scaleHeight(15) : scaleHeight(5)};
  width: 100%;
`;

const InputContainer = styled.View`
  justify-content: flex-end;
`;

const TextWrapper = styled(Text)`
  margin-bottom: ${scaleHeight(4)};
`;

const IconWrapper = styled.Image`
  height: ${scaleHeight(24)};
  position: absolute;
  right: ${scaleWidth(10)};
  width: ${scaleHeight(24)};
`;

const InputBoxContainer = styled.View`
  flex-direction: row;
  align-items: center;
`;

const InputFormBox = ({ title, error, value, onChangeText, isRequired, onEndEditing, ...rest }) => {
  const [wasOnFocus, setWasOnFocus] = useState(false);
  let borderColor = color.silverWhite;
  let borderWidth = 1;
  let font = fontFamily.calibreRegular;

  if (value) {
    font = fontFamily.calibreSemiBold;
  }

  if (wasOnFocus) {
    if (isRequired) {
      borderColor = !error ? color.greenValid : color.pinkRed;
      borderWidth = 2;
    }
  }

  return (
    <InputContainer {...rest}>
      <TextWrapper fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XSMALL} color={color.white}>
        {title}
      </TextWrapper>
      <InputBoxContainer>
        <InputBox
          allowFontScaling={false}
          testID="input-box"
          borderColor={borderColor}
          borderWidth={borderWidth}
          placeholder={rest.placeholder}
          placeholderTextColor={color.grayPlaceholder}
          onChangeText={onChangeText}
          onBlur={() => setWasOnFocus(true)}
          value={value}
          fontFamily={font}
          autoCapitalize="none"
          autoCorrect={false}
          onEndEditing={onEndEditing}
        />
        {wasOnFocus && isRequired && <IconWrapper source={error === null ? tick : errorIcon} />}
      </InputBoxContainer>
    </InputContainer>
  );
};

InputFormBox.defaultProps = {
  error: null,
  isRequired: true,
  value: null,
};

InputFormBox.propTypes = {
  error: PropTypes.string,
  isRequired: PropTypes.bool,
  onChangeText: PropTypes.func.isRequired,
  onEndEditing: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
};

export default InputFormBox;
