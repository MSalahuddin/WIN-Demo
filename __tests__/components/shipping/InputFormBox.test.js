import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import InputFormBox from '../../../js/components/shipping/InputFormBox';

describe('Component InputFormBox', () => {
  const onChangeText = jest.fn();

  test('renders correctly in default state', () => {
    const output = render(<InputFormBox title="a" error={null} value="a" onChangeText={onChangeText} />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when there is an error', () => {
    const output = render(<InputFormBox title="a" error="error" value="a" onChangeText={onChangeText} />);
    expect(output).toMatchSnapshot();
  });

  test('calls onChangeText when inputs are being entered', () => {
    const { getByTestId } = render(<InputFormBox title="a" error={null} value="a" onChangeText={onChangeText} />);
    fireEvent(getByTestId('input-box'), 'onChangeText', 'a');
    expect(onChangeText).toHaveBeenCalled();
  });
});
