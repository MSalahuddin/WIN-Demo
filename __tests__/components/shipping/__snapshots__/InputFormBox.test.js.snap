// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InputFormBox renders correctly in default state 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="REGULAR"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="XSMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreRegular",
          "fontSize": 24,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    a
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={false}
      autoCapitalize="none"
      autoCorrect={false}
      borderColor="#E5E5E5"
      borderWidth={1}
      fontFamily="Calibre-Semibold"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      placeholderTextColor="#ADADAD"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#E5E5E5",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "#222222",
            "fontSize": 21,
            "height": 70,
            "paddingLeft": 16,
            "paddingTop": 10,
            "width": "100%",
          },
        ]
      }
      testID="input-box"
      underlineColorAndroid="transparent"
      value="a"
    />
  </View>
</View>
`;

exports[`Component InputFormBox renders correctly when there is an error 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="REGULAR"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="XSMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreRegular",
          "fontSize": 24,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    a
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={false}
      autoCapitalize="none"
      autoCorrect={false}
      borderColor="#E5E5E5"
      borderWidth={1}
      fontFamily="Calibre-Semibold"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      placeholderTextColor="#ADADAD"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#E5E5E5",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "#222222",
            "fontSize": 21,
            "height": 70,
            "paddingLeft": 16,
            "paddingTop": 10,
            "width": "100%",
          },
        ]
      }
      testID="input-box"
      underlineColorAndroid="transparent"
      value="a"
    />
  </View>
</View>
`;
